### PR TITLE
Add overview of constant propagator

### PR DESCRIPTION
**Pull request by smowton**

### CODE REVIEW

File: src/analyses/README.md [hunk 1]
```diff
@@ -32,7 +32,10 @@ To be documented.
 
 \subsection analyses-constant-propagation Constant propagation (constant_propagator_ait)
 
-To be documented.
+A simple, unsound constant propagator. Replaces RHS symbol expressions (variable
+reads) with their values when they appear to have a unique value at a particular
+program point. Unsound with respect to pointer operations on the left-hand side
```
Comment (tautschnig) [inline src/analyses/README.md:37]: Is that genuinely the case? It's probably unsound for concurrent code, but I'm not sure about pointers.

File: src/analyses/README.md [hunk 2]
```diff
@@ -32,7 +32,10 @@ To be documented.
 
 \subsection analyses-constant-propagation Constant propagation (constant_propagator_ait)
 
-To be documented.
+A simple, unsound constant propagator. Replaces RHS symbol expressions (variable
```
Comment (peterschrammel) [inline src/analyses/README.md]: There should be a link to the file and class.

File: src/analyses/README.md (hunk 1 above)
Comment (smowton) [reply src/analyses/README.md:37]: Yes: https://github.com/diffblue/cbmc/blob/develop/src/analyses/constant_propagator.cpp#L38 (assign with anything but a symbol on the LHS is a no-op)

File: src/analyses/README.md (hunk 2 above)
Comment (smowton) [reply src/analyses/README.md]: Added (in heading)

**Review by chrisr-diffblue — APPROVED** [review]
LGTM

File: src/analyses/README.md (hunk 1 above)
Comment (tautschnig) [reply src/analyses/README.md:37]: Ouch. I wasn't quite aware how incomplete it was.

**Pull request merged**